web-ui/src/helpers/query-parameters.js

```diff
@@ -21,8 +21,21 @@ import { useEffect } from 'react';
 /**
  * @param {(QPBoolean | QPString)[]} qps - query parameters
  */
-export const useQueryParameters = qps => {
+export const useQueryParameters = (
+  qps,
+  requirements = [],
```

Comment:
Some pages require data to be loaded before query parameters can be processed. This page is one of them. We must have an array of all possible PDL objects in order to call `setSelectedPdls` with an array of PDL objects that have ids that match those in the `pdls` query parameter. Support this was extremely difficult and did not increase my love of React. ;-)

`requirements` is an array of data values that must be set in order to process the query parameters. It defaults to an empty array because most pages do not need this.

`processedQPs` is an object created by a call to `useRef` in the calling page. It holds a Boolean that indicates whether we have finished processing the query parameters. This can only happen after all the `requirements` values are present. I'm using a ref because the value needs to be maintained across multiple renders of the page which React loves to do. I'm open to suggestions for other ways to maintain this flag.